lib/travis/build/script/csharp.rb

```diff
@@ -59,6 +72,32 @@ def script
             sh.cmd 'false', echo: false, timing: false
           end
         end
+
+        def mono_repo
+          if config[:mono] == 'latest'
+            'wheezy'
+          else
+            if MONO_VERSION_REGEXP === config[:mono]
+              'wheezy/snapshots/' << config[:mono]
+            else
+              'wheezy'
+            end
+          end
+        end
+
+        def mono_version_valid?
+          if config[:mono] == 'latest' || MONO_VERSION_REGEXP === config[:mono]
+            if !is_mono_2? || (is_mono_2? && config[:mono] == '2.10.8')
+              true
+            end
+          end
+        end
```

Comment:
I think this whole method can be simplified to:

```ruby
return true if config[:mono] == 'latest'

if match = config[:mono].match(MONO_VERSION_REGEXP)
  major = match.captures
  return true if major >= 3 || (major == 2 && config[:mono] == '2.10.8')
end
```

do we want to factor the version check into a separate method?